changedetectionio/changedetection.py

```diff
@@ -20,13 +20,27 @@
 app = None
 datastore = None
 
-def sigterm_handler(_signo, _stack_frame):
+def graceful_shutdown():
     global app
     global datastore
-#    app.config.exit.set()
-    print('Shutdown: Got SIGTERM, DB saved to disk')
+    print('Gracefully exiting')
+    # Stop ChangeDetectionStore thread to avoid conflict with sync_to_json()
+    datastore.stop_thread = True
+    # Trigger saving data
     datastore.sync_to_json()
-#    raise SystemExit
+    # Stop check_for_new_version, notification_runner, ticker_thread_check_time_launch_checks, update_worker
+    app.config.exit.set()
+    sys.exit(0)
+
+def sigint_handler(sig, frame):
+    #https://docs.python.org/3/library/signal.html#signal.SIG_IGN
+    signal.signal(signal.SIGINT, signal.SIG_IGN)
```

Comment:
https://github.com/dgtlmoon/changedetection.io/pull/1850#issuecomment-1754794313 So since the SIGINT handler started, then ignore SIGINT.